### PR TITLE
fix(style): fix PasswordField and SearchField quiet styling

### LINKED ISSUE #1634: Primitives: Moving focus to TextField outerComponent causes 1px layout shift
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

Next

### Please describe your bug.

The fix for https://github.com/aws-amplify/amplify-ui/pull/1617 "fix: remove double border on TextField outer components" has a side effect of causing a 1px layout shift when focus moves to or from the button element in the outerComponent. This is most noticeable when the component is used inline, like on the home page of the docs.

layoutshift

An alternative considered for that PR was using a transparent border to "hide" the double border, though that causes a 1px gap in the top and bottom borders, which becomes even more apparent if your custom theme uses a wider border.

### What's the expected behaviour?

No layout shift occurs when moving focus to or from TextField outer components.

### Help us reproduce the bug!

You can reproduce on the docs home page by tabbing (or shift-tabbing) back and forth on the Search icon button in the Primitive Components section (in the gif above), or in an example app:

```
import { Flex, TextField, Button } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css'; // default theme

<Flex gap="20px" direction="row">
  <TextField label="Start" outerStartComponent={<Button>Start</Button>} />
  <TextField label="Default" />
</Flex>
```

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

_No response_

### LINKED ISSUE #1794: Bug: PasswordField / SearchField quiet variation button styling
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

React

### Which UI component?

Other

### How is your app built?

N/A

### Please describe your bug.

The styling of the show password button on the PassworField with `variation="quiet"` is off. The show password button shouldn't have a top and right border:

Screen Shot 2022-04-28 at 3 55 15 PM
 
Screen Shot 2022-04-28 at 5 01 19 PM

### What's the expected behaviour?

Show password button and search buttons should look nice.

### Help us reproduce the bug!

See: https://ui.docs.amplify.aws/components/passwordfield#variations

### Code Snippet

```typescript
// Put your code below this line.

```

### Additional information and screenshots

_No response_

### PR DESCRIPTION
**Pull request by zchenwei**
#### Description of changes

1. Update corresponding border colors to fix `PasswordField` and `SearchField` quiet variation styling. Make the colors `transparent` to avoid border pixel shift when switching between focus and non-focus state

2. Update unit tests to cover new cases.

#### Issue #, if available

Fixes #1794, fixes #1634

#### Description of how you validated changes

Screen Shot 2022-05-10 at 10 50 16 AM

Screen Shot 2022-05-10 at 10 50 25 AM

Screen Shot 2022-05-10 at 10 55 30 AM

Screen Shot 2022-05-10 at 10 55 41 AM

#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [x] Tests are updated
- [x] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dbanksdesign — APPROVED** [review]
LGTM! :shipit:

File: packages/ui/src/theme/css/component/fieldGroup.scss [hunk 1]
```diff
@@ -58,7 +58,7 @@
     @extend %remove-end-radii;
 
     &:not(:focus) {
-      border-inline-end: none;
+      border-inline-end-color: transparent;
```
Comment (reesscot) [inline packages/ui/src/theme/css/component/fieldGroup.scss:61]: Does this fix https://github.com/aws-amplify/amplify-ui/issues/1634?

**Review by reesscot — APPROVED** [review]
LGTM

**Review by wlee221 — APPROVED** [review]
🚢

File: packages/ui/src/theme/css/component/fieldGroup.scss (hunk 1 above)
Comment (zchenwei) [reply packages/ui/src/theme/css/component/fieldGroup.scss:61]: Oh yeah! It also fixes #1634. Thanks for bringing this up!

Comment (reesscot) [reply packages/ui/src/theme/css/component/fieldGroup.scss:61]: Want to add that to the description above so it's autoclosed?

Comment (zchenwei) [reply packages/ui/src/theme/css/component/fieldGroup.scss:61]: Yes will do

**Pull request merged**